index.js

```diff
@@ -1,10 +1,11 @@
 import filenamify from 'filenamify';
 import humanizeUrl from 'humanize-url';
+import is from '@sindresorhus/is';
 
-export default function filenamifyUrl(string, options) {
-	if (typeof string !== 'string') {
-		throw new TypeError('Expected a string');
+export default function filenamifyUrl(url, options) {
+	if (typeof url !== 'string' && !is.urlInstance(url)) {
```

Comment:
This doesn't really need adding a dependency.

```suggestion
	if (!(typeof url === 'string' || url instanceof URL)) {
```